.github/workflows/maven-publish.yml

```diff
@@ -0,0 +1,42 @@
+name: Publish snapshots to maven
+
+on:
+  workflow_dispatch:
+  push:
+      branches: [
+        main
+        1.*
+        2.*
+      ]
+
+jobs:
+  build-and-publish-snapshots:
+    strategy:
+      fail-fast: false
+      matrix:
+        jdk: [11, 17]
+        platform: ["ubuntu-latest", "windows-latest", "macos-latest"]
+    if: github.repository == 'opensearch-project/security'
+    runs-on: ${{ matrix.platform }}
+
+    permissions:
+      id-token: write
+      contents: write
+
+    steps:
+      - uses: actions/setup-java@v3
+        with:
+          distribution: temurin # Temurin is a distribution of adoptium
+          java-version: ${{ matrix.jdk }}
+      - uses: actions/checkout@v3
+      - uses: aws-actions/configure-aws-credentials@v1
+        with:
+          role-to-assume: ${{ secrets.PUBLISH_SNAPSHOTS_ROLE }}
+          aws-region: us-east-1
+      - name: publish snapshots to maven
+        run: |
+          export SONATYPE_USERNAME=$(aws secretsmanager get-secret-value --secret-id maven-snapshots-username --query SecretString --output text)
+          export SONATYPE_PASSWORD=$(aws secretsmanager get-secret-value --secret-id maven-snapshots-password --query SecretString --output text)
+          echo "::add-mask::$SONATYPE_USERNAME"
+          echo "::add-mask::$SONATYPE_PASSWORD"
+          ./gradlew publishMavenJavaPublicationToSnapshotsRepository
```

Comment:
This is the wrong publishing task, it creates the jars for upload, not the zip.

Reply:
Exactly. This is how it was handled before https://github.com/opensearch-project/opensearch-build/blob/main/scripts/default/opensearch/build.sh#L73-L83
And then maven cli was called via this script to publish everything https://github.com/opensearch-project/opensearch-build/blob/main/publish/publish-snapshot.sh
We need to figure out a way to do that directly via publishing task now.